### ISSUE TITLE
target blank to anchor

### ISSUE THREAD

**Issue by MartinMuzatko** [zengabor/zenscroll#19]
Hello!
Zenscroll works really great! I have one issue though. When I open a page with a location hash already in place, it does not account for the correct offset.

Is this something zenscroll will not take care of?

Thanks in advance

**Comment by zengabor** [comment]
When you open a page with a location hash then the browser does the positioning, zenscroll is not activated. If you want animated scrolling on the first page load then you need to use fake hashtags (hashtags that don't point to actual IDs on the page, e.g., "#go-mainarticle"), and a little script that detects and translates those fake hashes to animated scrolling to real IDs. Note, however, that in this case the page scrolling relies on your scripts and may not work for some users.

**Comment by MartinMuzatko** [comment]
I care more about the initial offset than the animation. Are there alternatives to this approach?

**Comment by zengabor** [comment]
For the offset you can add some margin or padding to the top of the target element, something like:

```
<div id="mainarticle" style="padding-top: 1em">
  <article>...
```

**Comment by MartinMuzatko** [comment]
@zengabor unfortunately, that won't work out for us, as this would destroy our vertical rhythm. So all I can do is manually check for `location.hash` and apply padding top appropriately.

So, there are no plans to take care of this? Or would this lead to unwanted behavior or is it snot part of the minimal approach of zenscroll?

Thanks for your reply.

**Comment by zengabor** [comment]
Sorry, Martin, this is indeed something beyond the scope of Zenscroll.

But you can include a one-liner into your script at the end of the body, if you are sure that zenscroll.js has been already loaded before that:

```
if (window.location.hash) zenscroll.to(document.getElementById(window.location.hash.substring(1)))
```

This will simply adjust the scroll by the specified offset. E.g., to have a 20 pixel offset, you put this before the one-liner: `zenscroll.setup(null, 20)`

You can also fine-tune how fast it should be (it can be instant if you specify `1` for duration as a second parameter in the one-liner).

The good thing about it as opposed to the fake IDs is that it doesn't break the default behavior of the browser but enhances it. :)

**Comment by zengabor** [comment]
Just realized that the one-liner above produces an error if the ID in the hash doesn't exist. So you should improve the script a little bit.

**Comment by MartinMuzatko** [comment]
Thank you @zengabor!

**Comment by zengabor** [comment]
This is now implemented in the 4.0 release candidate, which you can find here at the moment: https://github.com/zengabor/zenscroll/tree/gh-pages

Would you mind testing this functionality?

**Comment by MartinMuzatko** [comment]
Works as intended! thank you a lot :)
